PBN RCT.Rmd

```diff
@@ -59,114 +61,127 @@ included_participants <- merged_data %>%
 #Make necessary variables numeric
 included_participants <- included_participants %>%
   mutate(across(
-    starts_with(c("b_bhs_", "pi_bhs_", "b_shs_agency_", "pi_shs_agency_", "b_biss_", "pi_biss_", "b_fas_", "pi_fas_", "b_biaaq_", "pi_biaaq_", "b_chime_", "pi_chime_", "pi_pfs_1", "pi_pfs_2", "pi_pfs_3", "pi_pfs_4", "pi_pfs_5", "pi_pfs_6", "pi_pfs_7")),
+    starts_with(c("b_bhs_", "pi_bhs_", "b_shs_agency_", "pi_shs_agency_", "b_biss_", "pi_biss_", "b_fas_", "pi_fas_", "b_biaaq_", "pi_biaaq_", "b_chime_", "pi_chime_", "pi_pfs_1", "pi_pfs_2", "pi_pfs_3", "pi_pfs_4", "pi_pfs_5", "pi_pfs_6", "pi_pfs_7", "b_mfq_", "b_edeq_")),
     as.numeric
   ))
 
-#Create scores and filter for complete cases only
+#Create score and filter for program feedback completers
+included_participants_pfs <- included_participants %>%
+  mutate(pi_pfs_sum = pi_pfs_1 + pi_pfs_2 + pi_pfs_3 + pi_pfs_4 + pi_pfs_5 + pi_pfs_6 + pi_pfs_7) %>%
+  filter(!is.na(pi_pfs_sum))
+
+#Identify participants with greater than 5% missing baseline primary outcomes data
+#For each participant, add a new column that calculates the total number of baseline primary outcome items answered by each participant 
+included_participants <- included_participants %>%
+  mutate(total_responses_bprimary = rowSums(
+    !is.na(select(included_participants, b_mfq_1, b_mfq_2, b_mfq_3, b_mfq_4, b_mfq_5, b_mfq_6, b_mfq_7, b_mfq_8, b_mfq_9, b_mfq_10, b_mfq_11, b_mfq_12, b_mfq_13, b_edeq_1, b_edeq_2, b_edeq_3, b_edeq_4, b_edeq_5, b_edeq_6, b_edeq_7, b_edeq_8, b_edeq_9, b_edeq_10)), 
```

Comment:
Check out the helper functions in https://dplyr.tidyverse.org/reference/select.html, like `contains()`, which will save space here

Reply:
Gotcha. This could also save space

```
select(
  all_of(
    paste0("b_mfq_", 1:13)
  )
)
```

etc.